src/platform-includes/sourcemaps/upload/cli/javascript.mdx

```diff
@@ -24,30 +33,119 @@ For more info on Sentry CLI configuration visit the [Sentry CLI configuration do
+
+In order to use Debug IDs you will have to bundle your code with the `sourcesContent` field in the source map. This is required by Sentry when processing your events and will lead to problems in the case of absence since the context lines will not be computed.
+
+<Note>
+
+The generation of the `sourcesContent` will depend on your bundler configuration, therefore we reccomend you looking at the official docs.
+
+</Note>
+
+### 2. Enable The Debug IDs In Sentry CLI
+
+Since Debug IDs are currently opt-in only, you should enable the following environment variable, which will tell Sentry CLI to use the new Debug IDs system for uploading your artifacts.
+
+```bash
+export SENTRY_FORCE_ARTIFACT_BUNDLES=1
 ```
 
+### 3. Inject Debug IDs Using Sentry CLI
+
+Debug IDs require an additional but very important step, the injection. In this step, [`sentry-cli`](/product/cli/) will go over all the minified sources and their respective source maps and it will start adding custom metadata that is required by Sentry for correctly symbolicating the event.
+
+<Note>
+
+Debug IDs injection will modify the content of both your minified sources and source maps which will create issues if you have integrity checks in place or you check for file changes before upload (e.g., on a CDN) based uniquely on their filename. _We are currently working on improving the injection system in order to circumvent these problems._
+
+</Note>
+
+In order to inject Debug IDs you will have to use the new `inject` command:
+
 ```bash
-export SENTRY_AUTH_TOKEN=your-auth-token
-export SENTRY_ORG=___ORG_SLUG___
-export SENTRY_PROJECT=___PROJECT_SLUG___
+sentry-cli sourcemaps inject /path/to/file
 ```
 
-Then, create a release with Sentry CLI. The release name must be unique within your organization:
+<Note>
+
+You can verify your injected artifact files by looking at their contents and searching for `debugId` in the minified sources and `debug_id` in the source maps.
+
+</Note>
+
+### 4. Upload Artifacts Using Sentry CLI
+
+Once your artifact files are injected you can proceed to upload them to Sentry, with the `upload` command:
+
+```bash
+sentry-cli sourcemaps upload /path/to/file
+```
+
+<Note>
+
+You can verify your uploaded artifacts by navigating to **Project Settings -> Source Maps (left sidebar) -> Debug ID Bundles** on Sentry.
+
+</Note>
+
+<Note>
+
+Sentry CLI will notify you in case there are files that don't contain Debug IDs (e.g., you forgot to run the `inject` command). In that case, it will prompt you to weakly associate a release to the artifact. This weakly association is done in order to allow Sentry to properly symbolicate events without Debug IDs.
+
+</Note>
+
+#### Weak Release Association
+
+Since Debug IDs might cause issues for some users, we extended the new Debug IDs system with the concept of _Weak Release Association_. With such system you are able to tell Sentry to weakly associate the artifacts you upload with a `release`, and optionally `dist`.
+
```

Comment:
What problems? Good to be specific or don't say it at all as it sounds like the feature is half-baked.